### ISSUE TITLE
InitGooglePay Error

### ISSUE THREAD

**Issue by gwenoleR** [stripe/stripe-react-native#763]
**Describe the bug**
My GooglePlay console throw some errors due to this sdk.

```
java.lang.IllegalStateException: 
  at androidx.fragment.app.FragmentManager.checkStateLoss (FragmentManager.java:1844)
  at androidx.fragment.app.FragmentManager.enqueueAction (FragmentManager.java:1884)
  at androidx.fragment.app.BackStackRecord.commitInternal (BackStackRecord.java:329)
  at androidx.fragment.app.BackStackRecord.commit (BackStackRecord.java:294)
  at com.reactnativestripesdk.StripeSdkModule.initGooglePay (StripeSdkModule.kt:577)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.facebook.react.bridge.JavaMethodWrapper.invoke (JavaMethodWrapper.java:372)
  at com.facebook.react.bridge.JavaModuleWrapper.invoke (JavaModuleWrapper.java:151)
  at com.facebook.react.bridge.queue.NativeRunnable.run (Native Method)
  at android.os.Handler.handleCallback (Handler.java:900)
  at android.os.Handler.dispatchMessage (Handler.java:103)
  at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (MessageQueueThreadHandler.java:27)
  at android.os.Looper.loop (Looper.java:219)
  at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java:226)
  at java.lang.Thread.run (Thread.java:929)
```

**To Reproduce**
It looks like the error is from the native Google Pay init method, but I have no idea how to reproduce it. 
Here my code for the init method:

```js
if (isAndroid) {
      const { error } = await getStripe().initGooglePay({
        testEnv: false,
        merchantName: 'My merchant Name',
        countryCode: 'FR',
        billingAddressConfig: {
          format: 'FULL',
          isPhoneNumberRequired: true,
          isRequired: false,
        },
        existingPaymentMethodRequired: false,
        isEmailRequired: true,
      } as GooglePay.InitParams)

      if (error) {
        // DISCLAIMER: Should not block the flow if GooglePay is not available
        // throw error
        Sentry.addBreadcrumb({
          category: 'checkout',
          message: error?.message,
          data: error,
        })
      }
    }
```

**Expected behavior**
No crash when user try to init google pay

**Desktop (please complete the following information):**
 - OS: Android
 - RN version: 0.64.2
 - React: 17.0.1
 - Expo Version : SDK43
 - Stripe RN: 0.2.3

**Smartphone (please complete the following information):**
 - Device: Android
 - OS: All versions

**Comment by charliecruzan-stripe** [comment]
So this is error is only occurring for some of your users? Have you noticed a trend amongst the occurrences of the error (like location, OS version, or device type?)

Without the error or a way to reproduce the behavior, I'm not sure I can be of much help

**Comment by charliecruzan-stripe** [comment]
This issue might be alleviated (not sure since there's no way to reproduce it) by checking the `isGooglePaySupported` method first (released in v0.3.0)
